spring-cloud-gcp-pubsub/src/test/java/com/google/cloud/spring/pubsub/support/DefaultSubscriberFactoryTests.java

```diff
@@ -89,4 +126,49 @@ public void testCreatePullRequest_nonNullMaxMessages() {
 		PullRequest request = factory.createPullRequest("test", null, true);
 		assertThat(request.getMaxMessages()).isEqualTo(Integer.MAX_VALUE);
 	}
+
+	@Test
+	public void testGetExecutorProvider_customProvided() {
+		DefaultSubscriberFactory factory = new DefaultSubscriberFactory(() -> "project", null);
+		factory.setExecutorProvider(mockExecutorProvider);
+
+		assertThat(factory.getExecutorProvider("name"))
+				.isEqualTo(mockExecutorProvider);
+	}
+
+	@Test
+	public void testGetExecutorProvider_customNotProvided() {
+		DefaultSubscriberFactory factory = new DefaultSubscriberFactory(() -> "project", mockPubSubConfiguration);
+		when(mockPubSubConfiguration.getSubscriber("name")).thenReturn(new PubSubConfiguration.Subscriber());
+
+		assertThat(factory.getExecutorProvider("name"))
```

Comment:
Aha! This would be a good place to check that only one executor provider gets created for different subscriptions (unless there are customized properties).